src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+            propertiesBuilder.Add(OperationKey, operationKey);
+            propertiesBuilder.Add(IsAsyncKey, null);
+
+            if (shouldNegate)
+            {
+                propertiesBuilder.Add(ShouldNegateKey, null);
+            }
+
+            context.ReportDiagnostic(
+                operation.Syntax.CreateDiagnostic(
+                    rule: s_rule_CA1828,
+                    properties: propertiesBuilder.ToImmutable(),
+                    args: methodName));
+        }
+
+        private static void ReportCA1829(OperationAnalysisContext context, string propertyName, IOperation operation)
+        {
+            ImmutableDictionary<string, string?>.Builder propertiesBuilder = ImmutableDictionary.CreateBuilder<string, string?>();
+            propertiesBuilder.Add(PropertyNameKey, propertyName);
+
+            context.ReportDiagnostic(
+                operation.Syntax.CreateDiagnostic(
+                    rule: s_rule_CA1829,
+                    properties: propertiesBuilder.ToImmutable(),
+                    propertyName));
+        }
+
+        private static void ReportCA1836(OperationAnalysisContext context, bool useRightSideExpression, bool shouldNegate, IOperation operation)
+        {
+            ImmutableDictionary<string, string?>.Builder propertiesBuilder = ImmutableDictionary.CreateBuilder<string, string?>(StringComparer.Ordinal);
+
+            if (useRightSideExpression)
+            {
+                propertiesBuilder.Add(UseRightSideExpressionKey, null);
+            }
+
+            if (shouldNegate)
+            {
+                propertiesBuilder.Add(ShouldNegateKey, null);
+            }
+
+            context.ReportDiagnostic(
+                operation.Syntax.CreateDiagnostic(
+                    rule: s_rule_CA1836,
+                    properties: propertiesBuilder.ToImmutable()));
+        }
+
+        /// <summary>
+        /// Checks if the given methods is the <see cref="int.Equals(int)"/> methods.
+        /// </summary>
+        /// <param name="invocationOperation">The invocation operation.</param>
+        /// <returns><see langword="true"/> if the given methods is the <see cref="int.Equals(int)"/> methods; otherwise, <see langword="false"/>.</returns>
+        private static bool IsIntEqualsMethod(IInvocationOperation invocationOperation)
+        {
+            if (invocationOperation.Arguments.Length != 1)
+            {
+                return false;
+            }
+
+            IMethodSymbol methodSymbol = invocationOperation.TargetMethod;
+
+            return string.Equals(methodSymbol.Name, WellKnownMemberNames.ObjectEquals, StringComparison.Ordinal)
+                && IsInRangeInclusive((uint)methodSymbol.ContainingType.SpecialType, (uint)SpecialType.System_Int32, (uint)SpecialType.System_UInt64);
+        }
+
+        private static void AddIfNotNull(HashSet<IMethodSymbol> set, IEnumerable<IMethodSymbol>? others)
```

Comment:
Nit: make this a static local function for the single method where this is called?

Reply:
What's the benefit of it?